nixos/modules/services/cluster/patroni/default.nix

```
@@ -0,0 +1,249 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.patroni;
+  format = pkgs.formats.yaml { };
+
+  #boto doesn't support python 3.10 yet
+  patroni = pkgs.patroni.override { pythonPackages = pkgs.python39Packages; };
+
+  postgresql =
+    if cfg.postgresqlExtraPlugins == [ ]
+    then cfg.postgresqlPackage
+    else cfg.postgresqlPackage.withPackages (_: cfg.postgresqlExtraPlugins);
+
+  configFile = format.generate "patroni.yml" cfg.settings;
+in
+{
+  options.services.patroni = {
+
+    enable = mkEnableOption "Whether to enable Patroni";
+
+    postgresqlPackage = mkOption {
+      type = types.package;
+      example = literalExpression "pkgs.postgresql_11";
+      description = ''
+        PostgreSQL package to use.
+      '';
+    };
+
+    postgresqlExtraPlugins = mkOption {
+      type = types.listOf types.path;
+      default = [ ];
+      example = literalExpression "with pkgs.postgresql_11.pkgs; [ postgis pg_repack ]";
+      description = ''
+        List of PostgreSQL plugins. PostgreSQL version for each plugin should
+        match version for <literal>services.postgresql.package</literal> value.
+      '';
+    };
+
+    postgresqlDataDir = mkOption {
+      type = types.path;
+      defaultText = literalExpression ''"/var/lib/postgresql/''${config.services.patroni.postgresqlPackage.psqlSchema}"'';
+      example = "/var/lib/postgresql/14";
+      default = "/var/lib/postgresql/${postgresql.psqlSchema}";
+      description = ''
+        The data directory for PostgreSQL. If left as the default value
+        this directory will automatically be created before the PostgreSQL server starts, otherwise
+        the sysadmin is responsible for ensuring the directory exists with appropriate ownership
+        and permissions.
+      '';
+    };
+
+    postgresqlPort = mkOption {
+      type = types.int;
```

```suggestion
      type = types.port;
```
same for other port options.

---

Will do.